python/ray/_private/runtime_env/context.py

```diff
@@ -67,6 +67,14 @@ def exec_worker(self, passthrough_args: List[str], language: Language):
         else:
             executable = "exec "
 
+        default_worker_path = self.container.get("worker_path")
+        if self.container and default_worker_path:
+            logger.debug(
+                f"Changing the default worker path from {passthrough_args[0]} to "
+                f"{default_worker_path}."
+            )
+            passthrough_args[0] = default_worker_path
```

Comment:
I don't understand why it is needed here. can you comment why here?

Reply:
By default, raylet uses the path to `default_worker.py` on host. However, the path to `default_worker.py` inside the container started by podman can be different. This is how the original API works, but the `worker_path` value was being ignored in code; this part fixes it.

Reply:
Can we instead of overriding an argument so deep in the call-stack do it where invoke the worker instead?

Reply:
I think this is the location where we invoke the worker: https://github.com/ray-project/ray/blob/master/python/ray/_private/runtime_env/context.py#L85-L95